flexiblepricing/management/commands/create_product_for_program_courses.py

```diff
@@ -0,0 +1,85 @@
+
+from courses.models import CourseRun, Program
+from ecommerce.models import Product
+
+
+class Command(BaseCommand):
+    """
+    Create products for all the course runs in a program with standard price.
+    The default behavior is to create "inactive" product for each of DEDP courses with price $1000.
+    Unless specifying --active, running it more than once with default argument will create duplicated 'inactive' products
+    """
+
+    help = "Create products for each of course runs in a program with standard price"
+    PROGRAM_READABLE_ID = "program-v1:MITx+DEDP"
+    PRODUCT_PRICE = "1000"
+
+    def add_arguments(self, parser) -> None:
+        parser.add_argument(
+            "--program",
+            type=str,
+            help="The readable id of the program - e.g. program-v1:MITx+DEDP",
+            nargs="?",
+            default=self.PROGRAM_READABLE_ID,
+        )
+
+        parser.add_argument(
+            "--price",
+            type=str,
+            help="The standard price for each of the program courses. Default price is 1000",
+            nargs="?",
+            default=self.PRODUCT_PRICE,
+        )
+
+        parser.add_argument(
+            "--active",
+            action="store_true",
+            help="Create 'active' product. Default is 'inactive'",
+        )
+
+    def handle(self, *args, **options):
+        program_readable_id = options["program"]
+        price = options["price"]
+        active = True if options["active"] else False
+
+        program = Program.objects.filter(readable_id=program_readable_id).first()
+        if program is None:
+            raise CommandError(
+                f"Could not find program with readable_id - {self.PROGRAM_READABLE_ID}"
+            )
+
+        course_ids = program.courses.values_list("id", flat=True)
+        course_runs = CourseRun.objects.select_related("course").filter(
+            course_id__in=course_ids
+        )
+
+        self.stdout.write(
+            f"Creating product for each course in {program_readable_id} with price={price} and active={active})"
+        )
+
+        course_run_content_type = ContentType.objects.get(
+            app_label="courses", model="courserun"
+        )
+        for course_run in course_runs:
+            product, created = Product.objects.get_or_create(
+                object_id=course_run.id,
+                price=price,
+                content_type=course_run_content_type,
+                is_active=active,
+                defaults={"description": course_run.readable_id},
+            )
+
+            if created:
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        f"Created product for course run {course_run.readable_id}"
+                    )
+                )
+            else:
+                self.stdout.write(
+                    f"Product for course run - {course_run.readable_id} already exists - Skipping."
```

Comment:
get_or_create will only prevent duplicates for Product (object_id, is_active=True, content_type) here, not the case for is_active=False.
I can add active status to be clear

Reply:
I noticed when testing that if I have an active Product with a description such as "Collin's funky product" for a Course called "Collin's Course" which is part of "Collin's Program" and then I run the script with the `active` flag and the `program` flag specifying "Collin's Program", an exception will be thrown since there already exists a Product for Collin's Course.  It threw a `django.db.utils.IntegrityError: duplicate key value violates unique constraint "unique_purchasable_object"`

Reply:
@collinpreston I am not sure I follow this. The description shouldn't have anything to do with unique constrain. Did you have an active product for the same course run? unique_purchasable_object is for active  fields=["object_id", "is_active", "content_type"],

Reply:
Yes, I had an existing Product for the same CourseRun, but did not receive the error message on line 82.